### DISCUSSION TITLE
Blazor WebAssembly App with ASP.NET Core hosting - dotnet watch functionality?

### DISCUSSION THREAD

**Discussion by Waddlen** [dotnet/aspnetcore#25682] [General]
I greatly enjoyed `dotnet watch run` / `dotnet watch publish` when testing Blazor in VSC, but now I want to test doing API calls with an ASP.NET Core backend. So I created a Blazor WebAssemply app through Visual Studio 2019, like so:
![image](https://user-images.githubusercontent.com/23039119/92415388-abb46900-f126-11ea-8b0b-1455c3472641.png)

And this works, but annoyingly, the front-end parts of Blazor do not update when I refresh. So say I launch my app, add a word to `Index.razor`, then refresh => I don't see the change. I tried the "IIS Express", "TestApp1.Client", and "TestApp1.Server" launch settings the VS template shipped with, but none provide the `dotnet watch` functionality I want. I am using Chrome devtools with "Disable cache" checked.

Stopping/starting the app entirely is very slow and wastes a lot of time, is there a way to get `dotnet watch` functionality in this situation? Thanks!

**Comment by SteveSandersonMS** [comment]
We're greatly improving the functionality of `dotnet watch` in .NET Core 5.0. As far as I'm aware, it now correctly recompiles and refreshes any referenced project in your solution as of 5.0.

@pranavkm Is that correct?
